framework/harness/mock_endpoints.go

```diff
@@ -47,6 +47,34 @@ type MockEndpoint struct {
 	closing     sync.Once
 }
 
+type MockEndpointOption helpers.ConfigOption[MockEndpoint]
+
+type mockEndpointOptionContextFn struct {
+	contextFn func(context.Context) context.Context
```

Comment:
This one isn't new functionality, I just moved it from the regular parameters of NewMockEndpoint to the varargs options. Also, sdk-test-harness doesn't actually use this; it's used by sse-test-harness which shares this framework stuff.